AK/MaxElement.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Concepts.h>
+#include <AK/Iterator.h>
+
+namespace AK {
+
+template<typename TEndIterator, IteratorPairWith<TEndIterator> TIterator>
+constexpr auto max_element(
+    TIterator const& begin,
+    TEndIterator const& end,
+    auto const& comp)
+{
+    if (begin == end) {
+        return end;
+    }
+
+    TIterator largest = begin;
+    for (TIterator cur = begin + 1; cur != end; ++cur) {
+        if (comp(*largest, *cur)) {
+            largest = cur;
+        }
+    }
+    return largest;
+}
+
+template<typename TEndIterator, IteratorPairWith<TEndIterator> TIterator>
+constexpr auto max_element(
+    TIterator const& begin,
+    TEndIterator const& end)
+{
+    constexpr auto less = [](auto const& a, auto const& b) {
+        return a < b;
+    };
+    return max_element(begin, end, less);
+}
+
+template<IterableContainer Container>
+constexpr auto max_element(Container&& container, auto const& comp)
```

Comment:
I personally do not like the `auto` use for the return type and the comparator-function,
maybe force Containers or Iterators to expose the type with a `Container::Type` or just `Container::T`
(may require touching all common containers and their iterators)
as for the comparattor function, I think we use templates for Callbacks most of the time, maybe even make a Concept for it:
```cpp
template<typename OperandType, typename T>
concept Comparator = requires (T const& fun, OperandType const& op) { fun(op, op) -> bool };
```
(may require fixing up, it's been two months since I last touched concepts)